core/services/gateway/handlers/functions/orm.go

```diff
@@ -130,3 +134,37 @@ func (cs *cachedSubscriptionRow) encode() CachedSubscription {
 		},
 	}
 }
+
+func (o *orm) GetAllowedSenders(offset, limit uint, qopts ...pg.QOpt) ([]common.Address, error) {
+	var address []common.Address
+	stmt := fmt.Sprintf(`
+		SELECT allowed_address
+		FROM %s
+		WHERE router_contract_address = $1
+		ORDER BY id ASC
+		OFFSET $2
+		LIMIT $3;
+	`, allowlistTableName)
+	err := o.q.WithOpts(qopts...).Select(&address, stmt, o.routerContractAddress, offset, limit)
+	if err != nil {
+		return address, err
+	}
+
```

Comment:
Can we add a debug log message here indicating that we successfully fetched the allowlist from the DB? Include the offset, limit and length of `address`

Also, does the variable name `addresses` make more sense since it is an array of addresses?